code/game/atoms_movable.dm

```diff
@@ -24,12 +28,101 @@
+	var/area/current_gravity_area = null
 	var/floating = FALSE
 
+/atom/movable/vv_edit_var(var_name, var_value)
+	. = ..()
+	if(var_name == "is_affected_by_gravity")
+		sync_gravity()
+	if(var_name == "force_gravity_processing")
+		if(var_value)
+			atoms_forced_gravity_processing += src
+		else
+			atoms_forced_gravity_processing -= src
+
 /atom/movable/SDQL_update(const/var_name, new_value)
 	if(var_name == "step_x" || var_name == "step_y" || var_name == "step_size" || var_name == "bound_x" || var_name == "bound_y" || var_name == "bound_width" || var_name == "bound_height")
 		return FALSE	//PLEASE no.
 	. = ..()
+	if(var_name == "is_affected_by_gravity")
+		sync_gravity()
+	if(var_name == "force_gravity_processing")
+		if(new_value)
+			atoms_forced_gravity_processing += src
+		else
+			atoms_forced_gravity_processing -= src
+
+/atom/movable/proc/manual_gravity_process()
+	sync_gravity()
+	gravity_act()
+
+/atom/movable/proc/sync_gravity()
+	if(legacy_gravity)
+		return FALSE
+	if(gravity_ignores_area)
+		return FALSE
+	var/area/A = get_area(src)
+	if(!A)
+		return FALSE
+	if(current_gravity_area && (A != current_gravity_area))
+		stack_trace("[src] DID NOT UNREGISTER FROM [current_gravity_area] AND REGISTER TO [A] AFTER CHANGING AREAS!")
+		current_gravity_area.update_gravity(src, FALSE)
+		A.update_gravity(src, TRUE)
+	if(isturf(loc) || gravity_ignores_turfcheck)
+		if(forced_gravity_by_turf)
+			var/turf/T = loc
+			if(T != forced_gravity_by_turf)
+				stack_trace("[src] DID NOT UNREGISTER FROM [forced_gravity_by_turf] (TURF WITH MANUAL GRAVITY OVERRIDES)!")
+				forced_gravity_by_turf.reset_forced_gravity_atom(src)
+		else
+			A.update_gravity(src, is_affected_by_gravity)
+	else
+		if(forced_gravity_by_turf)
+			forced_gravity_by_turf.reset_forced_gravity_atom(src)
+		A.update_gravity(src, FALSE)	//We're not on a turf and we don't need to be included in processing.
+
+/atom/movable/Initialize()
+	var/turf/open/T = get_turf(src)
+	if(istype(T, /turf/open))
+		if(T.turf_gravity_overrides_area)
+			T.force_gravity_on_atom(src)
+	sync_gravity()
+	if(force_gravity_processing)
+		atoms_forced_gravity_processing += src
+	. = ..()
+
+/atom/movable/proc/gravity_act(moving = TRUE)
+	if(!isturf(src.loc))	//Gravity was so strong it was pulling shards and rods out of windows!
+		if(!gravity_ignores_turfcheck)	//We don't need to process.
+			sync_gravity()
+		return FALSE
+	if(anchored && !gravity_ignores_anchored)
+		return FALSE
+	if(!gravity_direction)
+		return FALSE
+	if(istype(get_step(src, gravity_direction), /turf/closed))
+		return FALSE
+	if(!override && !has_gravity())
+		return FALSE
+	if(!moving)
+		return TRUE
+	if(gravity_throwing && !throwing)
+		throw_at(get_edge_target_turf(get_turf(src), gravity_direction), gravity_strength * 20, gravity_strength * 2)
+		gravity_throwing = FALSE
+	else
+		for(var/i = gravity_strength, i > 0, i--)
+			Move(get_step(src, gravity_direction), gravity_direction)

```

Comment:
step() is just a caller for Move(). It's the same thing.

For future sanity though, please never use step(), ever.